SU2_CFD/src/solvers/CIncEulerSolver.cpp

```diff
@@ -450,9 +465,9 @@ void CIncEulerSolver::SetNondimensionalization(CConfig *config, unsigned short i
   config->SetDiffusivity_ConstantND(MassDiffusivityND);
 
 
-  /*--- Delete the original (dimensional) FluidModel object. No fluid is used for inscompressible cases. ---*/
+  /*--- Delete the original (dimensional) FluidModel object. No fluid is used for incompressible cases. ---*/
 
-  delete auxFluidModel;
+  if (redefine_fluid_model) delete auxFluidModel;
 
   /*--- Create one final fluid model object per OpenMP thread to be able to use them in parallel.
    *    GetFluidModel() should be used to automatically access the "right" object of each thread. ---*/
```

Comment:
Please don't re-use the auxFluidModel, is it too expensive to instantiate?
Looks like I need to fix this for coolprop :(

Reply:
For this one, it is fine to reuse the fluid model for the first entry in the vector of fluid models, for the others please make new fluid models otherwise this wont be thread safe.

Reply:
I have replaced fluidModel=auxFluidModel with fluidModel=new CFluidFlamelet(...)